vault/expiration.go

```diff
@@ -1269,6 +1274,17 @@ func (m *ExpirationManager) emitMetrics() {
 	num := len(m.pending)
 	m.pendingLock.RUnlock()
 	metrics.SetGauge([]string{"expire", "num_leases"}, float32(num))
+	// Check if lease count is greater than the threshold
+	if num > maxLeaseThreshold {
+		if atomic.LoadUint32(&m.leaseCheckCounter) > 59 {
+			m.logger.Warn("expiration: lease count exceeds maximum lease threshold")
+			atomic.StoreUint32(&m.leaseCheckCounter, 0)
+		} else {
+			atomic.AddUint32(&m.leaseCheckCounter, 1)
+		}
+	} else {
+		atomic.StoreUint32(&m.leaseCheckCounter, 0)
```

Comment:
This `else` clause is not necessary.

Reply:
There was an edge case where the counter will not get reset if the value was flapping over the value. But, thinking about it now, it is insignificant to warrant the else loop running all the time. I will remove it.

Reply:
It won't get reset, but I think that's totally fine. It won't display _more often_ than once every 60 seconds, it just may display less if it's flapping.